lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -417,11 +417,19 @@ public function processParameterValue($value)
             return $value;
         }
 
-        if (is_object($value) && $this->_em->getMetadataFactory()->hasMetadataFor(ClassUtils::getClass($value))) {
-            $value = $this->_em->getUnitOfWork()->getSingleIdentifierValue($value);
+        if (is_object($value)) {
+            try {
+                $this->_em->getMetadataFactory()->getMetadataFor(ClassUtils::getClass($value));
```

Comment:
Good code improvement indeed. I think the case is already tested under the "non-mapped class" comment with a \DateTime object.